xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-ui/src/main/resources/XWiki/Ratings.xml

```diff
@@ -220,16 +220,22 @@ document.observe("dom:loaded", function()
   {
      if(! el.hasClassName('locked')) 
      {
-        var url = "$xwiki.getURL("XWiki.Ratings", "view", "xpage=plain&amp;outputSyntax=plain")" + "&amp;doc=" + el.up('.rating-wrapper').id.substring(7);
-        var msgdiv = el.up('.rating-container').down('.rating-message');
+       var id = el.up('.rating-wrapper').id;
+       var PREFIX = 'rating-';
+       if (id &amp;&amp; id.length &gt; PREFIX.length)
+       {
+         id = id.substring(PREFIX.length);
+         var url = "$xwiki.getURL("XWiki.Ratings", "view", "xpage=plain&amp;outputSyntax=plain")" + "&amp;doc=" + encodeURIComponent(id);
```

Comment:
You don't need to use Velocity here. You can use the JavaScript API.

    new XWiki.Document('Ratings', 'XWiki').getURL(...)

Also, it's not nice to encode each parameter separately when you can use http://api.prototypejs.org/language/Object/toQueryString/ .

Reply:
Right, I used the dedicated function in the latest commit.